src/main/scala/beam/utils/scenario/UrbanSimScenarioLoader.scala

```diff
@@ -55,6 +57,31 @@ class UrbanSimScenarioLoader(
     if (wereCoordinatesInWGS) coord else geo.utm2Wgs(coord)
   }
 
+  private def buildAndAddLegToPlan(currentPlan: Plan, planElement: PlanElement): Leg = {
+    val leg = PopulationUtils.createAndAddLeg(currentPlan, planElement.legMode.getOrElse(""))
+    planElement.legDepartureTime.foreach(v => leg.setDepartureTime(v.toDouble))
+    planElement.legTravelTime.foreach(v => leg.setTravelTime(v.toDouble))
+    planElement.legMode.foreach(v => leg.setMode(v))
+    leg.getAttributes.putAttribute("trip_id", planElement.tripId)
```

Comment:
This is the only line it differs from `BeamScenarioLoader.buildAndAddLegToPlan`. I think we can reuse the same code.